### PR TITLE
Demonstration version for deployment

### PR DESCRIPTION
**Pull request by dave-promulgare**
Here are the contracts, server, and web application files.

### CODE REVIEW

File: wallet-rest-api/src/app.controller.ts [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import { Controller, Get } from '@nestjs/common';
```
Comment (mkhraisha) [inline wallet-rest-api/src/app.controller.ts:1]: Cleanup the hello world

File: wallet-rest-api/src/app.controller.spec.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
+import { Test, TestingModule } from '@nestjs/testing';
+import { AppController } from './app.controller';
+import { AppService } from './app.service';
+
```
Comment (mkhraisha) [inline wallet-rest-api/src/app.controller.spec.ts:4]: cleanup hello world

File: wallet-rest-api/src/app.service.ts [hunk 3]
```diff
@@ -0,0 +1,8 @@
+import { Injectable } from '@nestjs/common';
+
```
Comment (mkhraisha) [inline wallet-rest-api/src/app.service.ts:2]: cleanup hello world

File: wallet-rest-api/src/emission/emission.controller.ts [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+      console.log("getList query=", query);
+
+      const values = await this.emissionService.findAll(query)
+      response.set('Access-Control-Expose-Headers', 'X-Total-Count')
+      response.set('X-Total-Count', values.total)
+      response.status(200).send(values.page);
+
+    }
+    
+    @Get('getList')
```
Comment (mkhraisha) [inline wallet-rest-api/src/emission/emission.controller.ts]: should probably be 'list'

File: wallet-rest-api/src/emission/emission.controller.ts [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+        // get a single record by id
+    @Put(':id')
+    async putId(@Req() req): Promise<any> {
+        console.log("Params=", req.body);
+        let retval = await this.emissionService.putId(req.body);
+        console.log("Returns:", retval);
+        return retval;
+    }
+
+    @Post('getOne')
```
Comment (mkhraisha) [inline wallet-rest-api/src/emission/emission.controller.ts]: Remove the dummy gets?

File: wallet-rest-api/src/emission/emission.service.ts [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+    ) {}    
+
+    async create(createEmissionDto: CreateEmissionDto): Promise<Emissions> {
+        console.log("Create an emission database record");
+        const createdEmission = new this.emissionsModel(createEmissionDto);
+        return await createdEmission.save();
+      }
+    
+    async findAll(query: any): Promise<any> {
+        //query.id = query._id;
```
Comment (mkhraisha) [inline wallet-rest-api/src/emission/emission.service.ts]: remove commented line

File: wallet-rest-api/src/emission/emission.service.ts [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+        return emission;
+    }    
+
+    async putId(body: any): Promise<any> {
+        let emission = await this.emissionsModel.updateOne({_id: body.id}, body);
+        return body;
+    }    
+
+
+    async dummy(): Promise<any> {
```
Comment (mkhraisha) [inline wallet-rest-api/src/emission/emission.service.ts]: remove dummy

File: wallet-rest-api/src/oracle/oracle.controller.ts [hunk 8]
```diff
@@ -0,0 +1,90 @@
...
+    }
+
+    @Post('getOne')
+    getOne(): Promise<any> {
+        return this.oracleService.dummy();
+    }
+
+
+    // get a list of records based on an array of ids
+    @Post('getMany')
```
Comment (mkhraisha) [inline wallet-rest-api/src/oracle/oracle.controller.ts]: Remove dummies

File: wallet-rest-api/src/providers/database.providers.ts [hunk 9]
```diff
@@ -0,0 +1,9 @@
+import * as mongoose from 'mongoose';
+
+export const databaseProviders = [
+  {
+    provide: 'DATABASE_CONNECTION',
+    useFactory: (): Promise<typeof mongoose> =>
+      mongoose.connect('mongodb://192.168.2.192:27017/Mavennet'),
```
Comment (mkhraisha) [inline wallet-rest-api/src/providers/database.providers.ts]: make this an env var

File: wallet-rest-api/src/retire/retire.controller.ts [hunk 10]
```diff
@@ -0,0 +1,83 @@
+import { Controller, Get, Post, Body, Query, Res, Param } from '@nestjs/common';
+import { RetireService } from './retire.service';
+import { CreateRetireDto } from './dto/create-retire.dto';
+import { ApiTags } from '@nestjs/swagger';
+
+@ApiTags('Retire')
+@Controller('retire')
```
Comment (mkhraisha) [inline wallet-rest-api/src/retire/retire.controller.ts:7]: similar dummy comment

File: wallet-rest-api/src/retire/retire.service.ts [hunk 11]
```diff
@@ -0,0 +1,44 @@
...
+    ) {}    
+
+    async create(createRetireDto: CreateRetireDto): Promise<Retire> {
+        console.log("Create an emission retirement database record");
+        const createdRetire = new this.retireModel(createRetireDto);
+        return await createdRetire.save();
+    }
+    
+    async findAll(query: any): Promise<any> {
+        //query.id = query._id;
```
Comment (mkhraisha) [inline wallet-rest-api/src/retire/retire.service.ts]: remove line

File: wallet-rest-api/src/schemas/emissions.schema.ts [hunk 12]
```diff
@@ -0,0 +1,47 @@
...
+  @Prop()
+  emission_date: Date;
+
+  @Prop()
+  emission_type: Number;
+
+  @Prop()
+  emission_accuracy: Number;
+
+  //@Prop({ type: SchemaTypes.ObjectId, ref: 'Offsets', required:true })
```
Comment (mkhraisha) [inline wallet-rest-api/src/schemas/emissions.schema.ts]: delete unused code

**Review by mkhraisha — CHANGES_REQUESTED** [review]
Please cleanup hello world, dummy endpoints, and hardcoded strings.

File: wallet-rest-api/src/app.controller.ts (hunk 1 above)
Comment (dave-promulgare) [reply wallet-rest-api/src/app.controller.ts:1]: Fixed in latest version

File: wallet-rest-api/src/app.controller.spec.ts (hunk 2 above)
Comment (dave-promulgare) [reply wallet-rest-api/src/app.controller.spec.ts:4]: Fixed in latest version

File: wallet-rest-api/src/app.service.ts (hunk 3 above)
Comment (dave-promulgare) [reply wallet-rest-api/src/app.service.ts:2]: Fixed in latest version

File: wallet-rest-api/src/emission/emission.controller.ts (hunk 4 above)
Comment (dave-promulgare) [reply wallet-rest-api/src/emission/emission.controller.ts]: Removed

File: wallet-rest-api/src/emission/emission.controller.ts (hunk 5 above)
Comment (dave-promulgare) [reply wallet-rest-api/src/emission/emission.controller.ts]: Fixed in latest version

File: wallet-rest-api/src/emission/emission.service.ts (hunk 6 above)
Comment (dave-promulgare) [reply wallet-rest-api/src/emission/emission.service.ts]: Fixed in latest version

File: wallet-rest-api/src/emission/emission.service.ts (hunk 7 above)
Comment (dave-promulgare) [reply wallet-rest-api/src/emission/emission.service.ts]: Fixed in latest version

File: wallet-rest-api/src/oracle/oracle.controller.ts (hunk 8 above)
Comment (dave-promulgare) [reply wallet-rest-api/src/oracle/oracle.controller.ts]: Fixed in latest version

File: wallet-rest-api/src/providers/database.providers.ts (hunk 9 above)
Comment (dave-promulgare) [reply wallet-rest-api/src/providers/database.providers.ts]: Fixed in latest version

File: wallet-rest-api/src/retire/retire.controller.ts (hunk 10 above)
Comment (dave-promulgare) [reply wallet-rest-api/src/retire/retire.controller.ts:7]: Fixed in latest version

File: wallet-rest-api/src/retire/retire.service.ts (hunk 11 above)
Comment (dave-promulgare) [reply wallet-rest-api/src/retire/retire.service.ts]: Fixed in latest version

File: wallet-rest-api/src/schemas/emissions.schema.ts (hunk 12 above)
Comment (dave-promulgare) [reply wallet-rest-api/src/schemas/emissions.schema.ts]: Fixed in latest version

**Comment by dave-promulgare** [comment]
I have removed the Hello World call, the dummy calls, and the commented out code.  I have compiled and run the code and everything is working.

**Review by mkhraisha — APPROVED** [review]

**Pull request merged**